AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/MySqlBaseUserRepository.cs

```diff
@@ -37,4 +37,33 @@ public async Task<Model.BaseUser> CreateUser(string userId, string password)
             _settingsContext.BaseUsers.Single(u => u.UserId == userId)
         );
     }
+    public async Task<Model.BaseUser> DeleteUser(string userId)
+    {
+        var user = await _settingsContext.BaseUsers.SingleAsync(u => u.UserId == userId);
+
+        if (user != null)
+        {
+            _settingsContext.BaseUsers.Remove(user);
+            await _settingsContext.SaveChangesAsync();
+            return _baseUserConverter.Convert(user);
+        }
+        else{
+            return null;
+        }
```

Comment:
```suggestion
    _settingsContext.BaseUsers.Remove(user);
    await _settingsContext.SaveChangesAsync();
    return _baseUserConverter.Convert(user);
```

Not a huge deal but I would remove these null checks. .Single() will always either return a valid entry or throw an exception. We'll handle exceptions better later on but for now just letting the exception stop the execution seems efficient